doc/specs/#605 - Search/spec.md

```diff
@@ -0,0 +1,91 @@
+---
+author: Kaiyu Wang KaiyuWang16/kawa@microsoft.com
+created on: 2019-10-16
+last updated: 2019-10-16
+issue id: #605
+---
+
+# Search in Terminal
+
+## Abstract
+
+This spec is for feature request #605 "Search". It goes over the details of a new feature that allows users to search text in Terminal, within one tab or from all tabs. Expected behavior and design of this feature is included. Besides, future possible follow-up works are also addressed. 
+
+## Inspiration
+
+One of the superior features of iTerm2 is it's content search. The search comes in two variants: search from active tab and search from all tabs. In almost any editor, there is an roughly equivalent string search. We also want to realize search experience in Terminal. There will be two variants, search within one tab or from multiple tabs. We will start with one-tab search implementation. 
+
+## Solution Design
+
+Our ultimate goal is to provide both search within one tab and search from all tabs experiences. But we can start with one-tab search. The search experience should have following features:
+
+1. The search is triggered by keybindings "Ctrl + F". This coincides with other editors. In the future we will also consider adding a "Find" in the dropdown menu. 
+2. The user search in a XAML AutoSuggestBox on the top right corner of the Terminal window. 

```

Comment:
Why an `AutoSuggestBox`? Why not just a `TextBox`?

Reply:
I discussed with Kayla, and she would like the search box to be a AutoSuggest box because in the future we may consider auto suggest functionality.

Reply:
When we do have autocomplete suggestions, how will those appear with the "case sensitivity" checkbox beneath the search box? Will suggestions appear on top of the checkbox, partially obscuring it?